tests/cpp/test_multidevice_lower_communication.cpp

```diff
@@ -311,14 +310,14 @@ TEST_F(LowerCollectiveTest, ReduceScatter) {
   const auto device_id = communicator_->deviceId();
   at::Tensor in_tensor = shardTensor(unsharded_in_tensor, in, device_id);
 
-  FusionExecutor fe(communicator_);
-  fe.compileFusion(&fusion, {in_tensor});
-  assertIsCompiledToHostIrContainer(fe);
-  at::Tensor out_tensor = fe.runFusion({in_tensor})[0];
+  FusionExecutorCache fec(std::move(fusion), communicator_);
+  at::Tensor out_tensor = fec.runFusionWithInputs({in_tensor})[0];
+  assertIsCompiledToHostIrContainer(fec);
 
   at::Tensor unsharded_out_tensor = unsharded_in_tensor.sum(0);
   EXPECT_TRUE(at::allclose(
       out_tensor, shardTensor(unsharded_out_tensor, out, device_id)));
 }
 
+} // namespace
```

Comment:
is it on purpose that this namespace encompasses the whole file including the unit tests?

Reply:
Reverted.